src/handlers/pricing/action.ts

```diff
@@ -37,10 +37,53 @@ export const pricingLabelLogic = async (): Promise<void> => {
   const minPriorityLabel = priorityLabels.length > 0 ? priorityLabels.reduce((a, b) => (calculateWeight(a) < calculateWeight(b) ? a : b)).name : undefined;
 
   const targetPriceLabel = getTargetPriceLabel(minTimeLabel, minPriorityLabel);
+  let hasPriceLabel = false;
+  let hasTargetPriceLabel = false;
+
   if (targetPriceLabel) {
-    if (labels.map((i) => i.name).includes(targetPriceLabel)) {
-      logger.info(`Skipping... already exists`);
+    labelNames.forEach((e) => {
+      if (e.includes("Price")) {
+        hasPriceLabel = true;
+      }
+    });
+
+    labelNames.forEach((e) => {
+      if (e.includes(targetPriceLabel)) {
+        hasTargetPriceLabel = true;
+      }
+    });
+
+    if (hasPriceLabel) {
+      // below checks each label to see if theres a price
+      if (hasTargetPriceLabel) {
+        // get all issue events of type "labeled" and the event label includes Price
+        const labeledEvents = await getAllLabeledEvents();
+        if (!labeledEvents) return;
+        const labeledPriceEvents: typeof labeledEvents = [];
+        labeledEvents.forEach((event) => {
+          if (event.label?.name.includes("Price")) {
+            labeledPriceEvents.push(event);
+          }
+        });
+        // check if the latest price label has been added by a user
+        if (labeledPriceEvents[labeledPriceEvents.length - 1].actor?.type == "User") {
```

Comment:
add check for array length and you can use `UserType` enum instead of `User` literal